payments_service/braintree/forms.py

```diff
@@ -55,11 +55,18 @@ def clean(self):
                 forms.ValidationError('Unrecoginized plan_id: {}'
                                       .format(plan_id)))
 
-        if not self.user:
+        email = cleaned_data.get('email')
+        if self.user:
```

Comment:
I can't think of a way to do it that doesn't melt my brain because checking the email is independent -- the else block would be wrong if I combined it.